proxy/http/HttpTransact.cc

```diff
@@ -5795,11 +5799,8 @@ HttpTransact::initialize_state_variables_from_request(State *s, HTTPHdr *obsolet
   s->request_data.hostname_str = s->arena.str_store(host_name, host_len);
   ats_ip_copy(&s->request_data.src_ip, &s->client_info.src_addr);
   memset(&s->request_data.dest_ip, 0, sizeof(s->request_data.dest_ip));
-  if (s->state_machine->ua_session) {
-    NetVConnection *netvc = s->state_machine->ua_session->get_netvc();
-    if (netvc) {
-      s->request_data.incoming_port = netvc->get_local_port();
-    }
+  if (s->state_machine->ua_session && vc) {
```

Comment:
If ``vc`` is  not NULL then ``ua_session`` is also not NULL, so checking ``vc`` should be sufficient. Same for the previous check.